solidity/random-beacon/test/RandomBeacon.Relay.test.ts

```diff
@@ -164,27 +185,152 @@ describe("RandomBeacon - Relay", () => {
+                () => {
+                  let tx: ContractTransaction
+
+                  beforeEach(async () => {
+                    // We wait 20 blocks to make two more members eligible.
+                    // The member `18` submits the result.
+                    await mineBlocks(20)
+
+                    tx = await randomBeacon
+                      .connect(member18)
+                      .submitRelayEntry(
+                        firstEligibleMemberIndex + 2,
+                        blsData.groupSignature
+                      )
+                  })
+
+                  it("should remove members who did not submit from the sortition pool", async () => {
+                    await expect(tx)
+                      .to.emit(sortitionPool, "OperatorsRemoved")
+                      .withArgs([member16.address, member17.address])
+                  })
+
+                  it("should not slash any members", async () => {
+                    await expect(tx).to.not.emit(staking, "Slashed")
+                  })
+
+                  it("should emit RelayEntrySubmitted event", async () => {
+                    await expect(tx)
+                      .to.emit(randomBeacon, "RelayEntrySubmitted")
+                      .withArgs(1, blsData.groupSignature)
+                  })
+
+                  it("should terminate the relay request", async () => {
+                    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+                    expect(await randomBeacon.isRelayRequestInProgress()).to.be
+                      .false
+                  })
+                }
+              )
+
+              context(
+                "when first eligible member submits after the soft timeout",
+                () => {
+                  let tx: ContractTransaction
+                  let receipt: ContractReceipt
+
+                  beforeEach(async () => {
+                    // Let's assume we want to submit the relay entry after 75%
+                    // of the soft timeout period elapses. If so we need to
+                    // mine the following number of blocks:
+                    // `groupSize * relayEntrySubmissionEligibilityDelay +
+                    // (0.75 * relayEntryHardTimeout)`. However, we need to
+                    // subtract one block because the relay entry submission
+                    // transaction will move the blockchain ahead by one block
+                    // due to the Hardhat auto-mine feature.
+                    await mineBlocks(64 * 10 + 0.75 * 5760 - 1)
+
+                    tx = await randomBeacon
+                      .connect(member16)
+                      .submitRelayEntry(
+                        firstEligibleMemberIndex,
+                        blsData.groupSignature
+                      )
+
+                    receipt = await tx.wait()
+                  })
+
+                  it("should not remove any members from the sortition pool", async () => {
```

Comment:
Can we add another context "when some other than the first eligible member submits after the soft timeout" and ensure some operators are removed from the pool?

Reply:
Addressed in https://github.com/keep-network/keep-core/pull/2671/commits/e6ac2c4008995b5fb800737735d03c55061c3d03 being part of the follow-up PR: https://github.com/keep-network/keep-core/pull/2671